http/src/lib.rs

```diff
@@ -460,27 +461,25 @@ fn serve<M: jsonrpc::Metadata, S: jsonrpc::Middleware<M>>(
 			}
 		};
 
-		let handle = handle.clone();
 		bind_result.and_then(move |(listener, local_addr)| {
 			let allowed_hosts = server_utils::hosts::update(allowed_hosts, &local_addr);
 
-			let http = {
-				let mut http = server::Http::new();
-				http.keep_alive(keep_alive);
-				http.sleep_on_errors(true);
-				http
-			};
+			let mut http = server::conn::Http::new();
+			http.keep_alive(keep_alive);
+
 			listener.incoming()
-				.for_each(move |(socket, addr)| {
-					http.bind_connection(&handle, socket, addr, ServerHandler::new(
+				.for_each(move |socket| {
+					let service = ServerHandler::new(
 						jsonrpc_handler.clone(),
 						cors_domains.clone(),
 						cors_max_age,
 						allowed_hosts.clone(),
 						request_middleware.clone(),
 						rest_api,
 						max_request_body_size,
-					));
+					);
+					tokio::spawn(http.serve_connection(socket, service)
```

Comment:
Yes, in this case, because it is running within in the context of a future (which will be running on the executor we provided), there's no difference.